tools/debug-ui/src/BlocksView.tsx

```diff
@@ -0,0 +1,188 @@
+        case 'Error':
+            return 'Error';
+        default:
+            return '';
+    }
+}
+
+export const BlocksView = ({ addr }: BlocksViewProps) => {
+    const {
+        data: chainProcessingInfo,
+        error: chainProcessingInfoError,
+        isLoading: chainProcessingInfoLoading,
+    } = useQuery(['chainProcessingInfo', addr], () => fetchChainProcessingStatus(addr));
+    if (chainProcessingInfoLoading) {
+        return <div>Loading...</div>;
+    } else if (chainProcessingInfoError) {
+        return (
+            <div className="chain-and-chunk-blocks-view">
+                <div className="error">{(chainProcessingInfoError as Error).stack}</div>
+            </div>
+        );
+    } else if (!chainProcessingInfo) {
+        return (
+            <div className="chain-and-chunk-blocks-view">
+                <div className="error">No Data</div>
+            </div>
+        );
+    }
+    let numShards = 0;
+    numShards +=
+        chainProcessingInfo!.status_response.ChainProcessingStatus.blocks_info[0].chunks_info!
+            .length;
+    const shardIndices = [...Array(numShards).keys()];
+    return (
+        <div className="chain-and-chunk-blocks-view">
+            <table className="chain-and-chunk-blocks-table">
+                <thead>
+                    <tr>
+                        <th>Height</th>
+                        <th>Hash</th>
+                        <th>Received</th>
+                        <th>Status</th>
+                        <th>In Progress for</th>
+                        <th>In Orphan for</th>
+                        <th>Missing Chunks for</th>
+                        {shardIndices.map((shardIndex) => {
+                            return <th key={shardIndex}> Shard {shardIndex} </th>;
+                        })}
+                    </tr>
+                </thead>
+                <tbody>
+                    {chainProcessingInfo!.status_response.ChainProcessingStatus.blocks_info.map(
+                        (block) => {
+                            return (
+                                <tr key={block.height}>
+                                    <td>{block.height}</td>
+                                    <td>{block.hash}</td>
+                                    <td>{prettyTime(block.received_timestamp)}</td>
+                                    <td>{printBlockStatus(block.block_status)}</td>
+                                    <td>{block.in_progress_ms} ms</td>
+                                    <td>{printTimeInMs(block.orphaned_ms)}</td>
+                                    <td>{printTimeInMs(block.missing_chunks_ms)}</td>
+                                    {block.chunks_info!.map((chunk) => {
+                                        if (chunk) {
+                                            return (
+                                                <td key={chunk.shard_id}>
+                                                    <strong>
```

Comment:
- You can do something like ```<strong>{`${chunk.status} ${getChunkStatusSymbol(chunk.status)}`}</strong>``` so you can avoid having to use {' '} all the time
- `{x ? y : <></>}` can be simplified as `{x && y}`
- Do you really need to force spaces before <br/>?

Reply:
Updated all `{' '} {var} {' '}` to ```{` ${var} `}```
Updated all `{x ? y : <></>}` to `{x && y}` as suggested
A lot of these weird spaces and formatting were added by prettier from running:
`npx prettier --write src` because it complains from running `npx prettier --check src`